mesonbuild/msubprojects.py

```diff
@@ -15,23 +15,54 @@
 from .wrap.wrap import PackageDefinition, Resolver, WrapException, ALL_TYPES
 from .wrap import wraptool
 
+if T.TYPE_CHECKING:
+    from typing_extensions import Protocol
+
+    class Arguments(Protocol):
+        sourcedir: str
+        num_processes: int
+        subprojects: T.List[str]
+        types: str
+        # FIXME: Add narrowed type; descriptor protocol...
```

Comment:
Is this fixme still relavent?

Reply:
Kind of. This is reference to https://github.com/mesonbuild/meson/pull/9171#discussion_r753945340

Reply:
Notice the `type: ignore` in the PR.